src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -75,6 +80,9 @@ internal class BuildRequest
             var requestLength = args.Count + 1 + (libDirectory == null ? 0 : 1);
             var requestArgs = new List<Argument>(requestLength);
 
+            var compilerHash = GetCommitHash();
```

Comment:
Think we should make this a parameter. Every other part of the build request is an explicit value.